google-cloud-storage/lib/google/cloud/storage/file/signer_v4.rb

```diff
@@ -39,6 +39,60 @@ def self.from_bucket bucket, file_name
             new bucket.name, file_name, bucket.service
           end
 
+          ##
+          # The external path to the file, URI-encoded.
+          # Will not URI encode the special `${filename}` variable.
+          # "You can also use the ${filename} variable..."
+          # https://cloud.google.com/storage/docs/xml-api/post-object
+          #
+          def post_object_ext_path
+            path = "/#{@bucket_name}/#{@file_name}"
+            escaped = Addressable::URI.escape path
+            special_var = "${filename}"
+            # Restore the unencoded `${filename}` variable, if present.
+            if path.include? special_var
+              return escaped.gsub "$%7Bfilename%7D", special_var
+            end
+            escaped
+          end
+
+          def generate_signature signing_key, secret
```

Comment:
Commenting offline discussion around generate_signature:

The resulting signature must be in hex encoded format rather than base64 encoded format when using v4.

Reply:
Updated to hex with `hex_signature = signature.unpack('H*').first`. But the test is still failing with:

```
  1) Failure:
SignerV4PostObjectTest#test_bucket_0: POST Policy Simple [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-storage/test/google/cloud/storage/file/signer_v4_post_object_test.rb:87]:
--- expected
+++ actual
@@ -1 +1 @@
-"47ed5e45ea9bf72b889c0ba80b4424a0a8765d3c93558f4766e29a188c4ee70b7e5a287dc478e58087ae89e829324b0d3306d33bdd589e7092eb1b12a171023c37f28f2379f0d49eb9a9db62e3faa996ebff605a84318038fc4b2134532e50617709362b92f02cc9623acbc480dd964ee091121f8a5a5b074a625231ab507c6d223c09d727f56c5036ff56b719696a9958926274e7612e523716918aee351379feaffb92a6c111cee5d63d7d770fdb4a4e5d0f9e28971463cbf602015250e4b48e6822bc0cb550634da4b7d41e4d6caa817c7ed7bd4e30d87da81985cfc7eedee729b3221ff3b85f33efb92dfd24e7d5f6c135c1d93454a6364a1a4b8bf4b047"
+"2821a23ca8886ae9e9d67e8c094c06626a733f373cdca76d9d8c325e60ed8237b500b2f39937a15a1ebbdef7ed156fe3f41bc3456d453bb68d6755872eb429c329ae4e0dfb5e0abef4ca976394430ad7b3d1f14e50f6040c2cf81a2880980c26f9aeb800a674fc834b0e21e414adc3d1a0835da64107cce6a14bd2b39b4d6627029f821009e15ff3a7538ddc27dcf8c189ed197d25d50f0bddf58f99e0265adf562fff7f6689faba60920d73e6a6704c3d2be34a424766398396c55d3680f8dfd19513e2861fa4efa7ae99001877f043567132d413dc3c69e04a7ef5d30a325c2f9bc64e19dd6f0576781e546383378e7a4f2904411ee32d5c5a763683333dd7"
```